### ISSUE TITLE
Cannot execute query

### ISSUE THREAD

**Issue by mreid-moz** [mozilla/redash#451]
### Issue Summary

When clicking on the "Execute" button for [this query](https://sql.telemetry.mozilla.org/queries/56873/source#148630), the progress indicator does not appear, and the query execution never appears to complete.

### Steps to Reproduce

1. Open [the query](https://sql.telemetry.mozilla.org/queries/56873/source#148630)
2. Click "Execute"
3. See that the button becomes disabled, but the usual progress indicator does not appear.

I expected the "Executing..." UI to appear, with an indication of how long has passed since processing began.

I forked the query [here](https://sql.telemetry.mozilla.org/queries/57014/source#148916), and it seems to work as expected.

### Technical details:

* Redash Version: 4.0.1
* Browser/OS: Firefox Nightly 63 / OSX
* How did you install Redash: This is the redash deployment @ STMO

**Comment by jezdez** [comment]
Hm, this seems pretty weird, am I'm wondering if this is some kind of permission problem for queries that were created by other users. Definitely worth investigating.

**Comment by alison985** [comment]
I can reproduce this on https://sql.telemetry.mozilla.org/queries/56873/source#148629 I did notice that at the bottom where it previously stated a 5 minute runtime clicking the Execute button also changes the text to Running... runtime. I waited 10 minutes to see if it would update but it did not. The POST to events did fire but no others did. In looking at the code path when the Execute button is pressed I don't notice anything on a first pass that may be causing this. The next step is to dig deeper on any nuances of permissioning.

**Comment by alison985** [comment]
I can't reproduce this locally on the Mozilla fork code. I tried with 2 different user accounts to execute a query created by the 3rd user - one the initial admin account and another brand new account. This makes me think it's not a permissions issue but something else.

**Comment by mars-f** [comment]
Some more info: this also affects the query when it shows up on dashboards.  [This query](https://sql.telemetry.mozilla.org/queries/57184/source#149259) is infected.  The query is included in [this dashboard](https://sql.telemetry.mozilla.org/dashboard/engineering-workflow).  Pressing the Refresh button on the dashboard does not refresh the infected query panels.

**Comment by alison985** [comment]
I put my detective hat back on today and at least ruled some more things out. @jasonthomas I'm wondering if I can get server logs for the `worker` and/or `redis` around the string `Inserting job for ee090c3749a4fafbcd914a1ee78c0892`?

**Comment by jasonthomas** [comment]
I looked through the logs and I am not seeing anything in the scheduler or workers that matches `ee090c3749a4fafbcd914a1ee78c0892`. I tried digging into this more and looks like the `execute` action is being recorded in the `events` table for the problematic queries. I am not sure how this translates to scheduling the actual query to run but I assume that the scheduler/workers are not picking up these execute actions for some reason.

**Comment by alison985** [comment]
Thanks @jasonthomas.

@mreid-moz  Are there any other queries this is happening on? I can't reproduce it locally because table doesn't exist on my Athena connection.

**Comment by mreid-moz** [comment]
I don't have any other examples besides the ones I and mars-f posted here. I've asked on IRC and will update here if I find any more.

**Comment by rafrombrc** [comment]
We can't reproduce this locally, we have a workaround (forking the query) that unblocks the user in production. This may be a transient (caching, etc) error. Closing for now, please reopen if this resurfaces.

**Comment by hwine** [comment]
I'm experiencing the same issue on a regular basis. Forking isn't a viable solution, as that breaks all the dashboard, mana, etc links

I've reproduced as above using both Fx & Chrome as browsers.

One additional note: the query that hangs for me does have a refresh schedule enabled.

**Comment by hwine** [comment]
fwiw, I'm running a test assuming some interaction with the "refresh schedule" feature.
1. All my queries exhibiting the problem have a scheduled refresh time
2. A fork does not
3. Disabling the refresh time does not immediately fix the issue
4. Setting a refresh time on the fork does not immediately cause the issue

I set a refresh time on a [forked query], and will see what happens after the scheduled refresh occurs. Stay tuned....

[forked query]: https://sql.telemetry.mozilla.org/queries/60633/source

**Comment by washort** [comment]
Yes, I can immediately reproduce this by setting `schedule_resultset_size` to 1. I'll have a look as to why.

**Comment by washort** [comment]
https://github.com/mozilla/redash/pull/743
